cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -2211,6 +2250,110 @@ deserialiseFromRawBytesHex proxy hex =
       | otherwise        -> Nothing
 
 
+-- ----------------------------------------------------------------------------
+-- Bech32 Serialisation
+--
+
+class (HasTypeProxy a, SerialiseAsRawBytes a) => SerialiseAsBech32 a where
+  humanReadablePrefix :: a -> Bech32.HumanReadablePart
+
+  serialiseToBech32 :: a -> Either Bech32EncodeError Text
+  serialiseToBech32 a =
+    first Bech32EncodingError
+      . Bech32.encode (humanReadablePrefix a)
+      . Bech32.dataPartFromBytes
+      . serialiseToRawBytes
+      $ a
+
+  deserialiseFromBech32 :: AsType a -> Text -> Either Bech32DecodeError a
+  deserialiseFromBech32 asType bech32Str =
+      case Bech32.decode bech32Str of
```

Comment:
We'll need to the lenient variant, `Bech32.decodeLenient`. See https://github.com/input-output-hk/cardano-node/pull/1382#discussion_r449409435.